lib/collection/src/common/file_utils.rs

```diff
@@ -58,36 +58,57 @@ pub async fn move_file(from: impl AsRef<Path>, to: impl AsRef<Path>) -> Collecti
     Ok(())
 }
 
-/// Guard, that ensures that file will be deleted on drop.
-pub struct FileCleaner {
+/// Guard, ensures that a file or directory is deleted on drop.
+pub struct TempPath {
     path: Option<PathBuf>,
 }
```

Comment:
Why not use `tempfile::TempPath`/`tempfile::TempDir`, though?

Reply:
The idea was that our own structure would support deleting both file, directory and symlink paths. The `tempfile::TempPath` struct does not.

After a quick investigation it turns out that we never delete directories this way anymore, while we did that before.

I'll therefore refactor it out and switch to `tempfile::TempPath` instead.

Reply:
Done in 6b72e900b6baf6aec1853bb35f00404701d89740.